### ISSUE TITLE
[Windows-specific] Error on install: "Module parse failed" for font files

### ISSUE THREAD

**Issue by JangoBrick** [topcoder-platform/community-app#94]
The webpack build step during `npm install` takes a very long time (up to 20 minutes), requiring up to 4 GB of RAM, and then fails with hundreds of error messages similar to the following:

```
ERROR in ./src/assets/fonts/roboto/roboto-regular.woff
Module parse failed: D:\JangoBrick\GitHub\community-app\src\assets\fonts\roboto\roboto-regular.woff Unexpected character ' ' (1:4)
You may need an appropriate loader to handle this file type.
(Source code omitted for this binary file)
 @ ./~/css-loader?{"importLoaders":3,"localIdentName":"[path]___[name]__[local]___[hash:base64:5]","modules":true}!./~/postcss-loader/lib?{"plugins":[null]}!./~/resolve-url-loader!./~/sass-loader/lib/loader.js?{"sourceMap":true}!./src/shared/components/Dashboard/MyChallenges/ChallengeTile.scss 6:5667-5731

ERROR in ./src/assets/fonts/roboto/roboto-regular.ttf
Module parse failed: D:\JangoBrick\GitHub\community-app\src\assets\fonts\roboto\roboto-regular.ttf Unexpected character ' ' (1:0)
You may need an appropriate loader to handle this file type.
(Source code omitted for this binary file)
 @ ./~/css-loader?{"importLoaders":3,"localIdentName":"[path]___[name]__[local]___[hash:base64:5]","modules":true}!./~/postcss-loader/lib?{"plugins":[null]}!./~/resolve-url-loader!./~/sass-loader/lib/loader.js?{"sourceMap":true}!./src/shared/components/Dashboard/MyChallenges/ChallengeTile.scss 6:5763-5826

ERROR in ./src/assets/fonts/roboto/roboto-regular.svg
Module build failed: SyntaxError: D:/JangoBrick/GitHub/community-app/src/assets/fonts/roboto/roboto-regular.svg: Unexpected token (1:1)

> 1 | <?xml version="1.0" standalone="no"?>
    |  ^
  2 | <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd" >
  3 | <svg xmlns="http://www.w3.org/2000/svg">
  4 | <metadata></metadata>

 @ ./~/css-loader?{"importLoaders":3,"localIdentName":"[path]___[name]__[local]___[hash:base64:5]","modules":true}!./~/postcss-loader/lib?{"plugins":[null]}!./~/resolve-url-loader!./~/sass-loader/lib/loader.js?{"sourceMap":true}!./src/shared/components/Dashboard/MyChallenges/ChallengeTile.scss 6:5862-5925
```

... and so on.

I've done some research and found out that errors like that usually stem from missing/misconfigured rules for font files, which makes no sense since:

1. Webpack is clearly configured to handle them (see config/webpack/default.js):
    ```javascript
    {
      test: /\.(eot|otf|svg|ttf|woff|woff2)$/,
      include: [
        /src\/assets\/fonts/,
        /node_modules/,
      ],
      loader: 'file-loader',
      options: {
        outputPath: '/fonts/',
        publicPath: '/fonts/../',
      },
    }
    ```
2. The CircleCI builds do not fail.

I have retried `npm install` as well as `npm run build` several times to no success.
Does anybody else experience this and/or can help me with figuring out a fix?

Note: I am running Windows 10, Node v6.11.0, and npm v5.1.0.

**Comment by birdofpreyru** [comment]
@JangoBrick Everything builds fine on Ubuntu, I guess it is either Windows-specific problem, or something is wrong with your Node setup. What if you try to change `publicPath` in `file-loader` options to empty string (`''`)?

**Comment by JangoBrick** [comment]
@birdofpreyru Seems plausible, but does not appear to have changed anything.

You must be right about the problem being Windows-specific, it worked immediately on my Debian box (same Node/npm versions installed). I suppose I can set up a remote build system, so the issue is not as critical anymore, although it would still be nice to have it solved.

Thanks for your help though!

**Comment by birdofpreyru** [comment]
@JangoBrick Well, I keep this issue ticket open in case you, or anybody else, figures out what should be corrected in the config to work on Windows :)

**Comment by birdofpreyru** [comment]
@JangoBrick I have located the problem. The rule for font loader (and some other rules) in the Webpack config uses `/src\/assets\/fonts/` regular expression. It does not match assumed paths on Windows, as it uses different path separators :( I believe, the OS-independent way to write such regular expressions is `/src[\\/]assets[\\/]fonts/`. I'll test and push the update (probably won't have time to do it in the next couple of days, though).

There is another problem, that some npm scripts set up environment variables like `DEV_TOOLS=true node ./bin/www`, which is not supported on Windows. I don't know yet, what is the OS-independent way to do this, but we'll figure it out :)

**Comment by birdofpreyru** [comment]
Actually, I have pushed the update which is supposed to fix the Webpack build on Windows (edf205413a21360a6eca134af7d1250bb021bac0). Please tell whether it works for you, and feel free to suggest more changes to enhance the app compatibility with Windows (sure, they should not have any side-effects on Linux, which is anyway our primary target).

**Comment by JangoBrick** [comment]
@birdofpreyru Thanks a lot, I would not have expected you to be actively looking for a solution exactly because this is focused primarily on Linux anyway. I have checked and can confirm that your solution works flawlessly, so thanks again!

Closing this as the issue is fixed.
